### ISSUE TITLE
Cannot Compile on cygwin

### ISSUE THREAD

**Issue by tortured1** [flacon/shntool#1]
I tried to compile shntool on cygwin (both 32 and 64 bit), and it fails like this:

gcc -DHAVE_CONFIG_H -I. -I../include  -I../include  -I../include -g -O2 -MT mode_cat.o -MD -MP -MF .deps/mode_cat.Tpo -c -o mode_cat.o mode_cat.c
In file included from ../include/binary.h:15:0,
                 from ../include/mode.h:31,
                 from mode_cat.c:19:
mode_cat.c: In function ‘cat_file’:
../include/binary.h:80:63: error: ‘_O_BINARY’ undeclared (first use in this function); did you mean ‘O_BINARY’?
 # define SETBINARY_OUT(__fp)    (setmode   ( FILENO ((__fp)), _O_BINARY ))
                                                               ^
mode_cat.c:103:3: note: in expansion of macro ‘SETBINARY_OUT’
   SETBINARY_OUT(stdout);
   ^~~~~~~~~~~~~
../include/binary.h:80:63: note: each undeclared identifier is reported only once for each function it appears in
 # define SETBINARY_OUT(__fp)    (setmode   ( FILENO ((__fp)), _O_BINARY ))
                                                               ^
mode_cat.c:103:3: note: in expansion of macro ‘SETBINARY_OUT’
   SETBINARY_OUT(stdout);
   ^~~~~~~~~~~~~
make[1]: *** [Makefile:309: mode_cat.o] Error 1
make[1]: Leaving directory '/home/Greg/src/shntool/src'
make: *** [Makefile:251: all-recursive] Error 1

I traced the problem to binary.h:
#elif defined __CYGWIN__
# define SETBINARY_FD(__fp)     (setmode   ( (__fp), O_BINARY ))
# define SETBINARY_IN(__fp)     (setmode   ( FILENO ((__fp)), _O_BINARY ))
# define SETBINARY_OUT(__fp)    (setmode   ( FILENO ((__fp)), _O_BINARY ))

I changed _O_BINARY to O_BINARY:
#elif defined __CYGWIN__
# define SETBINARY_FD(__fp)     (setmode   ( (__fp), O_BINARY ))
# define SETBINARY_IN(__fp)     (setmode   ( FILENO ((__fp)), O_BINARY ))
# define SETBINARY_OUT(__fp)    (setmode   ( FILENO ((__fp)), O_BINARY ))

However, there's another issue:

gcc -I../include -g -O2   -o shntool.exe core_convert.o core_fileio.o core_format.o core_mode.o core_module.o core_output.o core_shntool.o core_wave.o glue_modes.o glue_formats.o mode_len.o mode_fix.o mode_hash.o mode_pad.o mode_join.o mode_split.o mode_cat.o mode_cmp.o mode_cue.o mode_conv.o mode_info.o mode_strip.o mode_gen.o mode_trim.o  format_wav.o format_aiff.o format_shn.o format_flac.o format_ape.o format_alac.o format_tak.o format_ofr.o format_tta.o format_als.o format_wv.o format_lpac.o format_la.o format_mkw.o format_bonk.o format_kxs.o format_cust.o format_term.o format_null.o
core_format.o: In function `spawn':
/home/Greg/src/shntool/src/core_format.c:134: undefined reference to `_open_osfhandle'
/home/Greg/src/shntool/src/core_format.c:135: undefined reference to `_open_osfhandle'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:256: shntool.exe] Error 1
make[1]: Leaving directory '/home/Greg/src/shntool/src'
make: *** [Makefile:251: all-recursive] Error 1

This is something of a known issue with cygwin, although I wasn't able to track down a solution...

**Comment by SokoloffA** [comment]
Unfortunately I do not support this program. Last years I don't use shntool in the [Flacon.](https://github.com/flacon/flacon) And I can't say anything about cygwin.
Perhaps makes sense for you to look at [cuetools](http://cue.tools/wiki/CUETools), if I'm not mistaken it has command line support.
